numpy/polynomial/_polybase.py

```diff
@@ -199,12 +199,7 @@ def has_samecoef(self, other):
             True if the coefficients are the same, False otherwise.
 
         """
-        if len(self.coef) != len(other.coef):
-            return False
-        elif not np.all(self.coef == other.coef):
-            return False
-        else:
-            return True
+        return len(self.coef) == len(other.coef) and np.all(self.coef == other.coef)
```

Comment:
Again, the original is easier to read.

Reply:
I've just split it into two lines for readability. However, if we use a formatter in the future, it would unfold again into a single line (because it wouldn't exceed the max line length).